source/common/common/json_escape_string.h

```diff
@@ -64,7 +64,7 @@ class JsonEscaper {
         position += 2;
         break;
       default:
-        if (character >= 0x00 && character <= 0x1f) {
+        if (character == 0x00 || (character > 0x00 && character <= 0x1f)) {
```

Comment:
So in the x86 platform, will the extra comparison come with a performance cost?

Reply:
Easy enough to verify using https://godbolt.org/ to examine the resulting cpu assembly code, would you mind checking that at -O2?

Reply:
Oh, the site works like magic. There is no overhead after compiler optimization.

Reply:
thanks for testing!

Reply:
for reference: https://godbolt.org/z/xrEfsd5Mq